File: src/Programmers/P1/P1_12928.java
package Programmers.P1;

import java.util.stream.Stream;

public class P1_12928 {
    public static void main(String[] args) {
        System.out.println(solution(10));
    }

    public static int solution(int n) {
        return Stream.iterate(1, v -> v + 1).limit(n).filter(v -> n % v == 0).reduce(Integer::sum).orElseGet(() -> 0);
    }
}

File: src/programmers/p1/P1_12928.kt
package programmers

fun main() {
    println(solution(0))
    println(solution(1))
    println(solution(2))
    println(solution(5))
    println(solution(12))
}

private fun solution(n: Int): Int = (1..n).filter {n % it == 0 }.sum()
private fun solution2(n: Int): Int = (1..n).filter { v -> n % v == 0 }.foldRight(0) { total, v -> total + v }